File: Examples/UIListView/src/course/examples/UI/ListLayout/ListViewActivity.java
package course.examples.UI.ListLayout;

import android.app.ListActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

public class ListViewActivity extends ListActivity {
	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		// Create a new Adapter containing a list of colors
		// Set the adapter on this ListActivity's built-in ListView
		setListAdapter(new ArrayAdapter<String>(this, R.layout.list_item,
				getResources().getStringArray(R.array.colors)));

		ListView lv = getListView();

		// Enable filtering when the user types in the virtual keyboard
		lv.setTextFilterEnabled(true);

		// Set an setOnItemClickListener on the ListView
		lv.setOnItemClickListener(new OnItemClickListener() {
			public void onItemClick(AdapterView<?> parent, View view,
					int position, long id) {

				// Display a Toast message indicting the selected item
				Toast.makeText(getApplicationContext(),
						((TextView) view).getText(), Toast.LENGTH_SHORT).show();
			}
		});
	}
}

File: ExamplesKotlin/UIListView/app/src/main/java/course/examples/ui/listlayout/ListViewActivity.kt
package course.examples.ui.listlayout

import android.app.ListActivity
import android.os.Bundle
import android.widget.AdapterView.OnItemClickListener
import android.widget.ArrayAdapter
import android.widget.TextView
import android.widget.Toast

class ListViewActivity : ListActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Create a new Adapter containing a list of colors
        // Set the adapter on this ListActivity's built-in ListView
        listAdapter = ArrayAdapter(
            this, R.layout.list_item,
            resources.getStringArray(R.array.colors)
        )

        val lv = listView

        // Enable filtering when the user types in the virtual keyboard
        lv.isTextFilterEnabled = true

        // Set an setOnItemClickListener on the ListView
        lv.onItemClickListener = OnItemClickListener { _, view, _, _ ->
            // Display a Toast message indicting the selected item
            Toast.makeText(
                applicationContext,
                (view as TextView).text, Toast.LENGTH_SHORT
            ).show()
        }
    }
}